compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/MetalRedrawer.kt

```diff
@@ -242,10 +269,7 @@ internal class MetalRedrawer(
         if (!isApplicationActive) {
             // If application goes background, synchronously schedule all inflightCommandBuffers, as per
             // https://developer.apple.com/documentation/metal/gpu_devices_and_work_submission/preparing_your_metal_app_to_run_in_the_background?language=objc
-            inflightCommandBuffers.forEach {
-                // Will immediately return for MTLCommandBuffer's which are not in `Commited` status
-                it.waitUntilScheduled()
-            }
+            inflightCommandBuffers.waitUntilAllAreScheduled()
```

Comment:
We probably should call this in `dispose` as well. Otherwise we can have a race between `dispose` and `encodeAndPresentBlock`

Reply:
They indeed can be executed in parallel, but I fail to see a problem in it, could you elaborate?

Reply:
1. dispose should end all background operations, because they consume resources, and can cause side effects
2. if we dispose something that is used in a background operation (pictureRecorder cache? context? can't say for sure, but better to be cautious), we can encounter a crash

Reply:
Hm, makes sense. But perhaps we need to do it the other way round and make `dispose` a critical section to be respected by a rendering thread?